site/docs/v1/tech/admin-guide/configuration-management/terraform.adoc

```diff
@@ -0,0 +1,502 @@
+  tenant_id = fusionauth_tenant.Default.id
+  name = "FusionAuth"
+}
+----
+Replace the 00000000-0000-0000-0000-000000000000 with your `FusionAuth Application Id`, you can find it in the `FusionAuth Admin UI` under `Applications`.
+
+Once appended run link:https://developer.hashicorp.com/terraform/cli/commands/plan[terraform plan] to check the validity of your configuration.
+
+If the plan is valid and you are happy with the changes run link:https://developer.hashicorp.com/terraform/cli/commands/apply[terraform apply]
+
+[source]
+----
+terraform plan
+terraform apply
+----
+
+[WARNING]
+====
+Once imported, Terraform tracks the resource in your state file. You can then manage the imported resource similar to any other by updating its attributes.
+
+With the exception of destroying the FusionAuth application, which normally works in Terraform lifecycle but as in FusionAuth this is a non-deletable application this wont work with Terraform either and will break your Terraform state. See <<Terraform Remove>> for more details.
+====
+
+We suggest to leave the import block in your configuration, a record of the resource's origin. The import block records that Terraform imported the resource and did not create it.
+
+=== Terraform Data Source Instead Of Import
+
+Instead of importing a resource you can use an existing FusionAuth configuration as a `Data Source`, the list of all supported data sources you can find in the link:https://registry.terraform.io/providers/gpsinsight/fusionauth/latest/docs[FusionAuth Terraform Provider Documentation]
+
+In the scenario where you choose to manage the Default tenant and FusionAuth application outside Terraform, but still add applications in to the default Tenant via Terraform, you can reference it in the configuration as a data source.
+
+* link:https://registry.terraform.io/providers/gpsinsight/fusionauth/latest/docs/data-sources/tenant[Terraform Data Source: Application]
+* link:https://registry.terraform.io/providers/gpsinsight/fusionauth/latest/docs/data-sources/application[Terraform Data Source: Application]
+
+[source]
+----
+data "fusionauth_tenant" "Default" {
+  name = "Default"
+}
+
+data "fusionauth_application" "FusionAuth" {
+  name = "FusionAuth"
+}
+----
+
+[NOTE.note]
+====
+If you manage the Default tenant and FusionAuth application outside of Terraform but want specific actions (scripts, API calls, etc.) still integrated and triggered by your Terraform configuration, you could make the use of link:https://developer.hashicorp.com/terraform/language/resources/provisioners/syntax[Provisioners].
+
+Terraform includes the concept of provisioners as a measure of pragmatism and last resort (Since provisioners are non-declarative and potentially unpredictable), knowing that there are always certain behaviors that cannot be directly represented in Terraform's declarative model.
+====
+//TODO Provisioner https://developer.hashicorp.com/terraform/language/resources/provisioners/syntax
```

Comment:
Sure. I left this as a placeholder, because we could make an example with a other tool managing something in FusionAuth outside of terraform. But then again the remark itself is a placeholder.